Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -26,6 +26,35 @@ extension MongoCollection {
         return try self._client.executeOperation(operation, session: session)
     }
 
+    /**
+     * Finds a single document in this collection that matches the provided filter.
+     *
+     * - Parameters:
+     *   - filter: A `Document` that should match the query
+     *   - options: Optional `FindOneOptions` to use when executing the command
+     *   - session: Optional `ClientSession` to use when executing this command
+     *
+     * - Returns:  the resulting `Document`, or nil if there is no match
+     *
+     * - Throws:
+     *   - `UserError.invalidArgumentError` if the options passed are an invalid combination.
+     *   - `UserError.logicError` if the provided session is inactive.
+     *   - `EncodingError` if an error occurs while encoding the options to BSON.
+     */
+
+    public func findOne(
+        _ filter: Document = [:],
+        options: FindOneOptions? = nil,
+        session: ClientSession? = nil
+    ) throws -> T? {
+        var findOneOptions: FindOptions?
```

Comment:
this could be condensed into a [`map`](https://developer.apple.com/documentation/swift/optional/1539476-map):
e.g.
```swift
let options = options.map { FindOptions(findOneOptions: $0) }
// or
let options = options.map(FindOptions.init(findOneOptions:))
```
I personally like using these functional thingies, but it's up to you which you think look better or prefer. I just figured I'd point out that something like this is possible.